.github/workflows/ci.yml

```diff
@@ -89,16 +102,30 @@ jobs:
           done
 
       # ────────────────────── Toolchain + Test Tools ─────────────────────────
-      - name: Run GCS feature tests
-        timeout-minutes: 5
+      - name: Run ${{ matrix.name }}
+        timeout-minutes: ${{ matrix.timeout-minutes }}
         run: |
-          cargo test -p moonlink --features=storage-gcs
+          ${{ matrix.cmd }}
 
-  # ──────────────────────── 3 · S3 feature test ────────────────────────────
+  # ──────────────────────── 3 · S3 test (feature, chaos) ───────────────────
   s3_test:
-    name: Unit tests with S3 feature enabled
+    name: S3 tests (feature, chaos)
     runs-on: ubuntu-latest
-
+    strategy:
+      matrix:
+        type: [feature, chaos]
+        include:
+          - type: feature
+            name: S3 feature test
+            timeout-minutes: 5
+            cmd: |
+              cargo test -p moonlink --features=storage-s3
+          - type: chaos
+            name: S3 chaos test
+            timeout-minutes: 10
+            cmd: |
+              cargo test storage::filesystem::accessor -p moonlink --features=chaos-test
+              RUST_BACKTRACE=1 cargo test table_handler::chaos_s3_test --features=chaos-test,storage-s3 -p moonlink -- --nocapture
```

Comment:
Could you please do me a favor, to switch `table_handler::chaos_s3_test` to `test_s3_chaos`?